packages/trader/src/Modules/Contract/Components/InfoBox/info-box-longcode.jsx

```diff
@@ -1,31 +1,31 @@
 import { observer } from 'mobx-react';
 import PropTypes from 'prop-types';
+import classNames from 'classnames';
 import React from 'react';
 import { Icon, Text } from '@deriv/components';
-import { localize, Localize } from '@deriv/translations';
-import { isVanillaContract } from '@deriv/shared';
-
-const generateMessageForVanillaTrade = contract_info => (
-    <Localize
-        i18n_default_text='<0>For {{contract_type}}:</0> Get a payout if {{index_name}} is {{strike_status}} than the strike price at the expiry time. Your payout is zero if the market is {{market_status}} or equal to the strike price at the expiry time. You will start making a profit when the payout is higher than your stake.'
-        components={[<strong key={0} />]}
-        values={{
-            contract_type: contract_info.contract_type === 'VANILLALONGCALL' ? localize('Call') : localize('Put'),
-            index_name: contract_info.display_name,
-            strike_status: contract_info.contract_type === 'VANILLALONGCALL' ? localize('higher') : localize('lower'),
-            market_status: contract_info.contract_type === 'VANILLALONGCALL' ? localize('lower') : localize('higher'),
-        }}
-    />
-);
+import { localize } from '@deriv/translations';
 
 const InfoBoxLongcode = ({ contract_info }) => {
-    const is_vanilla = isVanillaContract(contract_info.contract_type);
+    const max_longcode_length = 150;
+    const [is_collapsed, setIsCollapsed] = React.useState(true);
+
+    const handleToggle = () => {
+      setIsCollapsed(!is_collapsed);
+    };
+    
     return (
         <div className='info-box-longcode'>
             <Icon icon='IcContractFlag' className='info-box-longcode-icon' size={24} />
-            <Text size='xs' className='info-box-longcode-text'>
-                {is_vanilla ? generateMessageForVanillaTrade(contract_info) : contract_info.longcode}
-            </Text>
+            <div className='info-box-longcode-wrapper'>
+                <Text size='xs' className={classNames('info-box-longcode-text', {'info-box-longcode-text--collapsed' : is_collapsed})}>
+                    {contract_info.longcode}
+                </Text>
+                {` `}
```

Comment:
Isn't it better to add empty space with the help of CSS?

Reply:
I can only add left space to the button, since if I add it to description, it will add right space for each line, and not last word. In this case, if I add left space to the button, the space will also be visible in the beginning of the line whenever button starts on a new line, e.g collapsed mode, which is not needed behavior. It's also better for semantics, to add a space there, since it's a part of the same text block I think.

Reply:
Oh I see, but you can also add it inside of `{contract_info.longcode}`
Anyway, it doesn't seem critical, LGTM

Reply:
Missed your reply, yes, I first added inside the `Text` component, but due to jsx nature, it keeps trimming the empty space, therefore I added it explicitly. I also don't want to push it into the `contract_info.longcode` string as I think it might be very slightly less efficient as it executes an extra javascript calculation.